pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -635,3 +651,31 @@ func notFound(err error) bool {
 	}
 	return false
 }
+
+// generateValuesYaml generates a YAML string containing any
+// necessary values to override defaults in values.yaml. If
+// no defaults need to be overwritten, an empty string will
+// be returned.
+func generateValuesYaml() (string, error) {
+	httpProxy := os.Getenv("HTTP_PROXY")
+	httpsProxy := os.Getenv("HTTPS_PROXY")
+	noProxy := os.Getenv("NO_PROXY")
+
+	if httpProxy == "" && httpsProxy == "" && noProxy == "" {
+		// no proxy related environment variables set
+		return "", nil
+	}
+
+	proxySettings := proxyValues{
+		HttpProxy:  httpsProxy,
+		HttpsProxy: httpsProxy,
+		NoProxy:    noProxy,
+	}
+
+	yaml, err := yaml2.Marshal(proxySettings)
+	if err != nil {
+		return "", err
+	}
+
+	return string(yaml), nil
```

Comment:
What's the purpose in converting the resulting ```[]byte``` type to ```string```? I see the string used on lines 551 to 558, but am curious if the YAML bytestream could be used instead.

Reply:
It is because the valuesYAML field on app is a string, which is used to feed into the helm CLI.